### PR TITLE
Fixes issue #1954

### PR DESCRIPTION
**Pull request by trowacat**
See #1954
Links with tld of 2 chars is now being interpreted

### CODE REVIEW

File: src/addons/webLinks/webLinks.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ const protocolClause = '(https?:\\/\\/)';
 const domainCharacterSet = '[\\da-z\\.-]+';
 const negatedDomainCharacterSet = '[^\\da-z\\.-]+';
 const domainBodyClause = '(' + domainCharacterSet + ')';
-const tldClause = '([a-z\\.]{2,6})';
+const tldClause = '([a-z\\.]{1,6})';
```
Comment (Tyriar) [inline src/addons/webLinks/webLinks.ts]: Can you explain this change as I'm a bit puzzled as to why it wasn't working before. hostClause starts with:

```
const hostClause = '((' + domainBodyClause + '\\.' + tldClause + ')
```

Isn't tldClause matching only the "com" part from ".com"? I would have thought {2,6} would capture things like "io" and "com.au"?

Comment (trowacat) [reply src/addons/webLinks/webLinks.ts]: @Tyriar I thought this was odd as well but upon testing it appeared to fix the issue. I've looked deeper into it and have learned it was the pathClause which was causing the real problem in marking "io" or "com.au" as false.

Comment (Tyriar) [reply src/addons/webLinks/webLinks.ts]: Hmm, if we change pathClause like this we'll probably regress in not including the quotes or colon at the end of a URL (see https://github.com/xtermjs/xterm.js/commit/9cd7bf2af1178c29d6be4254bd13ed0c191ae299)

**Comment by Tyriar** [comment]
I looked into this a bit but I can't figure out how to do it without regressing in other important areas like supporting links like this without including the last semi-colon:

```
http://foo.bar:
http://foo.bar/:
http://foo.bar/foo:bar:
```

I think this is one of those issues that needs to get fixed when a non-regex link solution exists https://github.com/xtermjs/xterm.js/issues/583

**Pull request closed without merging**